/*
 * This file was automatically generated by EvoSuite
 * Sun Oct 02 22:24:35 KST 2016
 */


import static org.evosuite.runtime.EvoAssertions.assertThrownBy;
import static org.junit.Assert.fail;

import org.evosuite.runtime.EvoRunner;
import org.evosuite.runtime.EvoRunnerParameters;
import org.junit.Test;
import org.junit.runner.RunWith;

@RunWith(EvoRunner.class) @EvoRunnerParameters(useVNET = true) 
public class Tank_ESTest5   {

  @Test
  public void test00()  throws Throwable  {
      TankManager tankManager0 = null;
      int int0 = 0;
      Tank tank0 = null;
      try {
        tank0 = new Tank((TankManager) null, 21, 21, 21);
        fail("Expecting exception: NullPointerException");
      
      } catch(NullPointerException e) {
         //
         // no message in exception (getMessage() returned null)
         //
         assertThrownBy("Tank$$TankWar", e);
      }
  }

  @Test
  public void test01()  throws Throwable  {
      Tank tank0 = null;
      try {
        tank0 = new Tank((TankManager) null, 59, 59, 59);
        fail("Expecting exception: NullPointerException");
      
      } catch(NullPointerException e) {
         //
         // no message in exception (getMessage() returned null)
         //
         assertThrownBy("Tank$$TankWar", e);
      }
  }

  @Test
  public void test02()  throws Throwable  {
      Tank tank0 = null;
      try {
        tank0 = new Tank((TankManager) null, 23, 23, 23);
        fail("Expecting exception: NullPointerException");
      
      } catch(NullPointerException e) {
         //
         // no message in exception (getMessage() returned null)
         //
         assertThrownBy("Tank$$TankWar", e);
      }
  }

  @Test
  public void test03()  throws Throwable  {
      Tank tank0 = null;
      try {
        tank0 = new Tank((TankManager) null, 1017, 1017, 22);
        fail("Expecting exception: NullPointerException");
      
      } catch(NullPointerException e) {
         //
         // no message in exception (getMessage() returned null)
         //
         assertThrownBy("Tank$$TankWar", e);
      }
  }

  @Test
  public void test04()  throws Throwable  {
      Tank tank0 = null;
      try {
        tank0 = new Tank((TankManager) null, 20, 20, 20);
        fail("Expecting exception: NullPointerException");
      
      } catch(NullPointerException e) {
         //
         // no message in exception (getMessage() returned null)
         //
         assertThrownBy("Tank$$TankWar", e);
      }
  }

  @Test
  public void test05()  throws Throwable  {
      Tank tank0 = null;
      try {
        tank0 = new Tank((TankManager) null, 18, 18, 18);
        fail("Expecting exception: NullPointerException");
      
      } catch(NullPointerException e) {
         //
         // no message in exception (getMessage() returned null)
         //
         assertThrownBy("Tank$$TankWar", e);
      }
  }

  @Test
  public void test06()  throws Throwable  {
      Tank tank0 = null;
      try {
        tank0 = new Tank((TankManager) null, 20, 20, 17);
        fail("Expecting exception: NullPointerException");
      
      } catch(NullPointerException e) {
         //
         // no message in exception (getMessage() returned null)
         //
         assertThrownBy("Tank$$TankWar", e);
      }
  }

  @Test
  public void test07()  throws Throwable  {
      Tank tank0 = null;
      try {
        tank0 = new Tank((TankManager) null, 16, 16, 16);
        fail("Expecting exception: NullPointerException");
      
      } catch(NullPointerException e) {
         //
         // no message in exception (getMessage() returned null)
         //
         assertThrownBy("Tank$$TankWar", e);
      }
  }

  @Test
  public void test08()  throws Throwable  {
      Tank tank0 = null;
      try {
        tank0 = new Tank((TankManager) null, 15, 15, 15);
        fail("Expecting exception: NullPointerException");
      
      } catch(NullPointerException e) {
         //
         // no message in exception (getMessage() returned null)
         //
         assertThrownBy("Tank$$TankWar", e);
      }
  }

  @Test
  public void test09()  throws Throwable  {
      Tank tank0 = null;
      try {
        tank0 = new Tank((TankManager) null, (-2), 14, 14);
        fail("Expecting exception: NullPointerException");
      
      } catch(NullPointerException e) {
         //
         // no message in exception (getMessage() returned null)
         //
         assertThrownBy("Tank$$TankWar", e);
      }
  }

  @Test
  public void test10()  throws Throwable  {
      Tank tank0 = null;
      try {
        tank0 = new Tank((TankManager) null, 14, 13, 13);
        fail("Expecting exception: NullPointerException");
      
      } catch(NullPointerException e) {
         //
         // no message in exception (getMessage() returned null)
         //
         assertThrownBy("Tank$$TankWar", e);
      }
  }

  @Test
  public void test11()  throws Throwable  {
      Tank tank0 = null;
      try {
        tank0 = new Tank((TankManager) null, 20, 20, 12);
        fail("Expecting exception: NullPointerException");
      
      } catch(NullPointerException e) {
         //
         // no message in exception (getMessage() returned null)
         //
         assertThrownBy("Tank$$TankWar", e);
      }
  }

  @Test
  public void test12()  throws Throwable  {
      Tank tank0 = null;
      try {
        tank0 = new Tank((TankManager) null, 11, 11, 11);
        fail("Expecting exception: NullPointerException");
      
      } catch(NullPointerException e) {
         //
         // no message in exception (getMessage() returned null)
         //
         assertThrownBy("Tank$$TankWar", e);
      }
  }

  @Test
  public void test13()  throws Throwable  {
      Tank tank0 = null;
      try {
        tank0 = new Tank((TankManager) null, 10, 628, 10);
        fail("Expecting exception: NullPointerException");
      
      } catch(NullPointerException e) {
         //
         // no message in exception (getMessage() returned null)
         //
         assertThrownBy("Tank$$TankWar", e);
      }
  }

  @Test
  public void test14()  throws Throwable  {
      Tank tank0 = null;
      try {
        tank0 = new Tank((TankManager) null, 9, 9, 9);
        fail("Expecting exception: NullPointerException");
      
      } catch(NullPointerException e) {
         //
         // no message in exception (getMessage() returned null)
         //
         assertThrownBy("Tank$$TankWar", e);
      }
  }

  @Test
  public void test15()  throws Throwable  {
      Tank tank0 = null;
      try {
        tank0 = new Tank((TankManager) null, 33, 8, 8);
        fail("Expecting exception: NullPointerException");
      
      } catch(NullPointerException e) {
         //
         // no message in exception (getMessage() returned null)
         //
         assertThrownBy("Tank$$TankWar", e);
      }
  }

  @Test
  public void test16()  throws Throwable  {
      Tank tank0 = null;
      try {
        tank0 = new Tank((TankManager) null, 7, 7, 7);
        fail("Expecting exception: NullPointerException");
      
      } catch(NullPointerException e) {
         //
         // no message in exception (getMessage() returned null)
         //
         assertThrownBy("Tank$$TankWar", e);
      }
  }

  @Test
  public void test17()  throws Throwable  {
      Tank tank0 = null;
      try {
        tank0 = new Tank((TankManager) null, 6, 6, 6);
        fail("Expecting exception: NullPointerException");
      
      } catch(NullPointerException e) {
         //
         // no message in exception (getMessage() returned null)
         //
         assertThrownBy("Tank$$TankWar", e);
      }
  }

  @Test
  public void test18()  throws Throwable  {
      Tank tank0 = null;
      try {
        tank0 = new Tank((TankManager) null, 0, 5, 5);
        fail("Expecting exception: NullPointerException");
      
      } catch(NullPointerException e) {
         //
         // no message in exception (getMessage() returned null)
         //
         assertThrownBy("Tank$$TankWar", e);
      }
  }

  @Test
  public void test19()  throws Throwable  {
      Tank tank0 = null;
      try {
        tank0 = new Tank((TankManager) null, 4, 4, 4);
        fail("Expecting exception: NullPointerException");
      
      } catch(NullPointerException e) {
         //
         // no message in exception (getMessage() returned null)
         //
         assertThrownBy("Tank$$TankWar", e);
      }
  }

  @Test
  public void test20()  throws Throwable  {
      Tank tank0 = null;
      try {
        tank0 = new Tank((TankManager) null, (-14), 3, 3);
        fail("Expecting exception: NullPointerException");
      
      } catch(NullPointerException e) {
         //
         // no message in exception (getMessage() returned null)
         //
         assertThrownBy("Tank$$TankWar", e);
      }
  }

  @Test
  public void test21()  throws Throwable  {
      Tank tank0 = null;
      try {
        tank0 = new Tank((TankManager) null, 2, 2, 2);
        fail("Expecting exception: NullPointerException");
      
      } catch(NullPointerException e) {
         //
         // no message in exception (getMessage() returned null)
         //
         assertThrownBy("Tank$$TankWar", e);
      }
  }

  @Test
  public void test22()  throws Throwable  {
      Tank tank0 = null;
      try {
        tank0 = new Tank((TankManager) null, 1, 1, 1);
        fail("Expecting exception: NullPointerException");
      
      } catch(NullPointerException e) {
         //
         // no message in exception (getMessage() returned null)
         //
         assertThrownBy("Tank$$TankWar", e);
      }
  }

  @Test
  public void test23()  throws Throwable  {
      Tank tank0 = null;
      try {
        tank0 = new Tank((TankManager) null, 0, 0, 0);
        fail("Expecting exception: NullPointerException");
      
      } catch(NullPointerException e) {
         //
         // no message in exception (getMessage() returned null)
         //
         assertThrownBy("Tank$$TankWar", e);
      }
  }

  @Test
  public void test24()  throws Throwable  {
      Tank tank0 = null;
      try {
        tank0 = new Tank((TankManager) null, 19, 19, 19);
        fail("Expecting exception: NullPointerException");
      
      } catch(NullPointerException e) {
         //
         // no message in exception (getMessage() returned null)
         //
         assertThrownBy("Tank$$TankWar", e);
      }
  }
}
